openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -461,6 +463,21 @@ private static List<GenSrcFile> generateClientFiles(OASClientConfig oasClientCon
         String licenseContent = oasClientConfig.getLicense();
         BallerinaClientGenerator ballerinaClientGenerator = getClientGenerator(oasClientConfig, statusCodeBinding);
         String mainContent = Formatter.format(ballerinaClientGenerator.generateSyntaxTree()).toString();
+
+        List<ClientDiagnostic> clientDiagnostic = ballerinaClientGenerator.getDiagnostics();
+
+        if (!clientDiagnostic.isEmpty()) {
+            for (ClientDiagnostic diagnostic : clientDiagnostic) {
+                createDiagnostics(toolContext, diagnostic.getMessage(), diagnostic.getCode(),
+                        diagnostic.getDiagnosticSeverity(), location);
+            }
+
+            if (clientDiagnostic.stream().anyMatch(
+                    diagnostic -> diagnostic.getDiagnosticSeverity() == DiagnosticSeverity.ERROR)) {
+                throw new ClientException("Error occurred while generating client");
+            }
+        }
```

Comment:
Cant we simply do
```suggestion
            for (ClientDiagnostic diagnostic : clientDiagnostic) {
                createDiagnostics(toolContext, diagnostic.getMessage(), diagnostic.getCode(),
                        diagnostic.getDiagnosticSeverity(), location);
            }

            if (clientDiagnostic.stream().anyMatch(
                    diagnostic -> diagnostic.getDiagnosticSeverity() == DiagnosticSeverity.ERROR)) {
                throw new ClientException("Error occurred while generating client");
            }
```
?

Reply:
Fixed via [b32c1ac](https://github.com/ballerina-platform/openapi-tools/pull/1693/commits/b32c1ac0aa57f2ec3550d3ca18e4dea1e3dc5309)